torchdata/dataloader2/dataloader2.py

```diff
@@ -62,7 +72,15 @@ def __next__(self) -> T_co:
                 if self.dataloader:
                     self.dataloader.shutdown()
                 raise
-        else:
+            finally:
```

Comment:
I feel there is no need to do `finally`. You can move those lines to the line before `return next_val`.

Reply:
Would it be better to returning the value to the model first before executing pause? Or you think it will make 0 difference at all?

Reply:
It might be my PTSD caused by the racing generator problem 😢 TBH I don't have a strong opinion on it